contracts/MigrationContractStaging.cdc

```diff
@@ -0,0 +1,362 @@
+    ///
+    access(all) view fun getAllStagedContractCode(forAddress: Address): {String: String} {
+        if self.stagedContracts[forAddress] == nil {
+            return {}
+        }
+        let capsulePaths: [StoragePath] = []
+        let stagedCode: {String: String} = {}
+        let contractNames: [String] = self.stagedContracts[forAddress]!
+        for name in contractNames {
+            capsulePaths.append(self.deriveCapsuleStoragePath(contractAddress: forAddress, contractName: name))
+        }
+        for path in capsulePaths {
+            if let capsule = self.account.borrow<&Capsule>(from: path) {
+                let update: ContractUpdate = capsule.getContractUpdate()
+                stagedCode[update.name] = update.codeAsCadence()
+            }
+        }
+        return stagedCode
+    }
+
+    /// Returns a StoragePath to store the Capsule of the form:
+    ///     /storage/self.capsulePathPrefix_ADDRESS_NAME
+    access(all) view fun deriveCapsuleStoragePath(contractAddress: Address, contractName: String): StoragePath {
+        return StoragePath(
+                identifier: self.capsulePathPrefix
+                    .concat(self.delimiter)
+                    .concat(contractAddress.toString())
+                    .concat(self.delimiter)
+                    .concat(contractName)
+            ) ?? panic("Could not derive Capsule StoragePath for given address")
+    }
+
+    /* ------------------------------------------------------------------------------------------------------------ */
+    /* ------------------------------------------------ Constructs ------------------------------------------------ */
+    /* ------------------------------------------------------------------------------------------------------------ */
+
+    /********************
+        ContractUpdate
+     ********************/
+
+    /// Represents contract and its corresponding code.
+    ///
+    access(all) struct ContractUpdate {
+        access(all) let address: Address
+        access(all) let name: String
+        access(all) var code: String
+
+        init(address: Address, name: String, code: String) {
+            self.address = address
+            self.name = name
+            self.code = code
+        }
+
+        /// Validates that the named contract exists at the target address.
+        ///
+        access(all) view fun isValid(): Bool {
+            return getAccount(self.address).contracts.names.contains(self.name)
+        }
+
+        /// Serializes the address and name into a string of the form 0xADDRESS.NAME
+        ///
+        access(all) view fun toString(): String {
+            return self.address.toString().concat(".").concat(self.name)
+        }
+
+        /// Returns human-readable string of the Cadence code.
+        ///
+        access(all) view fun codeAsCadence(): String {
+            return String.fromUTF8(self.code.decodeHex()) ?? panic("Problem stringifying code!")
```

Comment:
Code should be staged as-is, as a `String`, there is no need to keep it hex-encoded.

The hex-encoding used to be done done when sending the code in a transaction, because how inefficient it is to send a `[UInt8]`, but even that is no longer necessary, see https://github.com/onflow/sdks/issues/17.

Simply remove the function and use the `code` field.

Reply:
> Code should be staged as-is

Do you mean that code should be staged as human-readable Cadence?